src/components/App.js

```diff
@@ -1,39 +1,55 @@
 import React from 'react';
 import { hot } from 'react-hot-loader/root';
-import { checkURIforError, getTokenFromURI, getAuthURL } from '../config/authConfig';
-import macaroon from '../assets/img/Macaroonicon.png';
+import { checkURIforError, getTokenFromURI, getTokenFromLocal } from '../config/authConfig';
+import { Dashboard } from './Dashboard';
+import { Sidebar } from './Sidebar';
+import { Welcome } from './Welcome';
+import { LoginLink } from './LoginLink';
+import UserPlaylists from './UserPlaylists';
+import CurrentPlaylist from './CurrentPlaylist';
 
 
 class App extends React.Component {
   state = {
     token: '',
-    user: '',
-    error: false,
-    errorMessage: '',
+    user: {
+      name: '',
+      id: '',
+      error: false,

```

Comment:
I'd move `error` and `errorMessage` up one level. There are other network requests you might make that may require error handling, and it's usually easier to manage those at a more global level rather than a localized one.

For bigger apps making multiple network requests to multiple endpoints (sometimes in parallel), it's key to know which ones failed and which one didn't, so in those cases localized error handling would be more appropriate.

Reply:
Right, I noticed that too just haven't gotten around to fixing.